pages/api/roles/index.ts

```diff
@@ -1,5 +1,129 @@
+    [{ type: StatusMessageType.Error, message: 'The company for this role does not exists.' }],
+  ],
+  [MessageType.EMPTY_TITLE, [{ type: StatusMessageType.Error, message: 'Role title is empty.' }]],
+  [MessageType.INVALID_TYPE, [{ type: StatusMessageType.Error, message: 'Role type is invalid.' }]],
+  [MessageType.INVALID_YEAR, [{ type: StatusMessageType.Error, message: 'Role year is invalid.' }]],
+  [
+    MessageType.ROLE_CREATED_SUCCESSFULLY,
+    [{ type: StatusMessageType.Success, message: 'Role was created succesfully.' }],
+  ],
+]);
+
+function handler(req: NextApiRequest, res: NextApiResponse) {
+  const method = req.method;
+
+  switch (method) {
+    case HTTP_GET_METHOD:
+      handleGet(req, res);
+      break;
+    case HTTP_POST_METHOD:
+      handlePost(req, res);
+      break;
+    default:
+      rejectHttpMethod(res, method);
+  }
+}
+
+// TODO: Optional Search query. Will be done in another PR to unblock others.
+
+async function handleGet(_: NextApiRequest, res: NextApiResponse<ApiResponse<RoleListData[]>>) {
+  const roles: RoleListData[] = await prisma.role.findMany({
+    select: {
+      id: true,
+      title: true,
+      type: true,
+      year: true,
+      isVerified: true,
+      company: { select: { id: true, name: true, companyUrl: true } },
+    },
+  });
+
+  res.status(HTTP_STATUS_OK).json(createJsonResponse(roles));
 }
+
+async function handlePost(req: NextApiRequest, res: NextApiResponse<ApiResponse<RoleData | EmptyPayload>>) {
+  if (!(await isValidRequest(req, res))) {
+    return;
+  }
+
+  const rolePostData: RolePostData = req.body;
+
+  createIfPossible(res, async () => {
+    const newRole: RoleData = await prisma.role.create({
+      data: rolePostData,
+      select: { id: true, title: true, type: true, year: true },
+    });
+
+    res
+      .status(HTTP_STATUS_CREATED)
+      .json(createJsonResponse(newRole, messages.get(MessageType.ROLE_CREATED_SUCCESSFULLY)));
+  });
+}
+
+async function isValidRequest(req: NextApiRequest, res: NextApiResponse<ApiResponse<EmptyPayload>>): Promise<boolean> {
+  if (typeof req.body.companyId !== 'number') {
+    res.status(HTTP_STATUS_BAD_REQUEST).json(createJsonResponse({}, messages.get(MessageType.INVALID_COMPANY_ID)));
+    return false;
+  }
+
+  const company = await prisma.company.findFirst({ where: { id: req.body.companyId } });
+
+  if (!company) {
+    res.status(HTTP_STATUS_BAD_REQUEST).json(createJsonResponse({}, messages.get(MessageType.COMPANY_DOES_NOT_EXISTS)));
+    return false;
+  }
+
+  if (isEmpty(req.body.title)) {
+    res.status(HTTP_STATUS_BAD_REQUEST).json(createJsonResponse({}, messages.get(MessageType.EMPTY_TITLE)));
+    return false;
+  }
+
+  if (isEmpty(req.body.type) || !Object.values(RoleType).includes(req.body.type)) {
+    res.status(HTTP_STATUS_BAD_REQUEST).json(createJsonResponse({}, messages.get(MessageType.INVALID_TYPE)));
+    return false;
+  }
+
+  if (typeof req.body.year !== 'number') {
```

Comment:
Right now any number is okay?

Reply:
Yes. I think @ganhongyao wanted didn't want to add a CHECK constraint so that people can track past applications.

Reply:
But I can put -1 and that would also work?